tests/models/test_hparams.py

```diff
@@ -538,5 +513,6 @@ def test_args(tmpdir):
     trainer.fit(model)
 
     raw_checkpoint_path = _raw_checkpoint_path(trainer)
-    model = SubClassVarArgs.load_from_checkpoint(raw_checkpoint_path)
-    assert model.hparams == hparams
+    with pytest.raises(TypeError, match="__init__\(\) got an unexpected keyword argument 'test'"):
+        SubClassVarArgs.load_from_checkpoint(raw_checkpoint_path)
```

Comment:
not sure why this test shall pass before if the hparams was assigned by force...  #2911

Reply:
@jbschiratti I wrote this test and now I was thinking about this trying to figure out what my fix was here, but I simply cannot remember. 
Is it working for inheritance as expected? Maybe you understand it better